modules/raft/src/main/java/org/apache/ignite/raft/jraft/storage/snapshot/local/LocalSnapshotCopier.java

```diff
@@ -260,6 +260,10 @@ private void loadMetaTable() throws InterruptedException {
                 setError(session.status().getCode(), session.status().getErrorMsg());
                 return;
             }
+            // No need in further snapshot processing if it was already cancelled and the metaBuf is null.
+            if (metaBuf.getBuffer() == null && session.status().isCanceled()) {
+                return;
+            }
             if (!this.remoteSnapshot.getMetaTable().loadFromIoBufferAsRemote(metaBuf.getBuffer())) {
```

Comment:
if (!this.remoteSnapshot.getMetaTable().loadFromIoBufferAsRemote(metaBuf.getBuffer()) && isOk())